### ISSUE TITLE
wrong output about PyTorch Kernels in the user guide

### ISSUE THREAD

**Issue by brabbitdousha** [shader-slang/slang#3430]
I followed the user guide in [https://shader-slang.com/slang/user-guide/a1-02-slangpy.html#getting-started-with-slangpy](https://shader-slang.com/slang/user-guide/a1-02-slangpy.html#getting-started-with-slangpy)
However, when it comes to **Wrapping your kernels as pytorch functions**
the ```square.slang``` provided above seems wrong for the dim mentioned in the pytorch code:

```
[AutoPyBindCUDA]
[CUDAKernel]
[Differentiable]
void square(DiffTensorView input, DiffTensorView output)
{
    uint3 dispatchIdx = cudaThreadIdx() + cudaBlockIdx() * cudaBlockDim();

    if (dispatchIdx.x >= input.size(0))
        return;
    
    output[dispatchIdx.x] = input[dispatchIdx.x] * input[dispatchIdx.x];
}
```
pytorch code
```
kernel_with_args.launchRaw(
            blockSize=(32, 32, 1),
            gridSize=((input.shape[0] + 31) // 32, (input.shape[1] + 31) // 32, 1))
```
follow codes provided, I can only get 
```
dX = tensor([[6., 0.],
        [0., 0.]])
```
However, in the guide,:
```
dX = tensor([[6., 8.],
        [0., 2.]])
```

then I changed the code to be along with the square.slang above:
```
import torch
import slangpy

m = slangpy.loadModule('square.slang')

class MySquareFunc(torch.autograd.Function):
    @staticmethod
    def forward(ctx, input):
        output = torch.zeros([4],dtype=torch.float).to('cuda:0')
        kernel_with_args = m.square(input=input, output=output)
        kernel_with_args.launchRaw(
            blockSize=(4, 1, 1),
            gridSize=(1, 1, 1))
        ctx.save_for_backward(input, output)
        return output
    
    @staticmethod
    def backward(ctx, grad_output):
        (input, output) = ctx.saved_tensors
        input_grad = torch.zeros_like(input)
        # Note: When using DiffTensorView, grad_output gets 'consumed' during the reverse-mode.
        # If grad_output may be reused, consider calling grad_output = grad_output.clone()
        #
        kernel_with_args = m.square.bwd(input=(input, input_grad), output=(output, grad_output))
        kernel_with_args.launchRaw(
            blockSize=(4, 1, 1),
            gridSize=(1, 1, 1))
        
        return input_grad

    def test_backward(input, output):
        input_grad = torch.zeros_like(input)
        grad_output = torch.ones_like(output) 
        # Note: When using DiffTensorView, grad_output gets 'consumed' during the reverse-mode.
        # If grad_output may be reused, consider calling grad_output = grad_output.clone()
        #
        kernel_with_args = m.square.bwd(input=(input, input_grad), output=(output, grad_output))
        kernel_with_args.launchRaw(
            blockSize=(4, 1, 1),
            gridSize=(1, 1, 1))
        
        return input_grad
    
x = torch.tensor([3.0, 4.0, 0.0, 1.0], dtype=torch.float, requires_grad=True, device='cuda')
x_copy = x.clone()
print(f"X = {x}")
y_pred = MySquareFunc.apply(x)
loss = y_pred.sum()
loss.backward()
print(f"dX = {x.grad.cpu()}")
print(MySquareFunc.test_backward(x_copy, y_pred))
```
However, I still get 
```
dX = tensor([[6., 0.],
        [0., 0.]])
```
But my test_backward gets the right thing, could you give me any hint of it ?

**Comment by csyonghe** [comment]
It seems the shader is only processing the first dimension of the tensors. Fixed the script to pass in one dimensional tensors.
